homeassistant/components/timer/__init__.py

```diff
@@ -385,9 +386,9 @@ def async_cancel(self):
         self.async_write_ha_state()
 
     @callback
-    def async_finish(self):
+    def async_finish(self) -> None:
         """Reset and updates the states, fire finished event."""
-        if self._state != STATUS_ACTIVE:
+        if self._state != STATUS_ACTIVE or self._end is None:
             return
```

Comment:
And here too

Reply:
Same here. Once `self._state` is set to `STATUS_ACTIVE`, `self._end` is also set.
https://github.com/home-assistant/core/blob/1d2c23d81c56a38944a77d3537c599136cfa9cc3/homeassistant/components/timer/__init__.py#L315-L324